### PR TITLE
Create wip

### PR DESCRIPTION
**Pull request by carkom**
Adding WIP to new PRs

### CODE REVIEW

File: upcoming-release-notes/2443.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
```
Comment (MatissJanis) [inline upcoming-release-notes/2443.md:2]:
```suggestion
category: Maintenance
```

File: .github/workflows/wip.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Add WIP
+
+on:
+  pull_request:
+    types:
+      - opened
+
+jobs:
+  greeting:
```
Comment (MatissJanis) [inline .github/workflows/wip.yml:9]:
```suggestion
  add_wip_prefix:
```

File: .github/workflows/wip.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+    - name : Add WIP
```
Comment (MatissJanis) [inline .github/workflows/wip.yml:14]:
```suggestion
    - name: Add WIP
```

**Review by MatissJanis — COMMENTED** [review]
Will this work for PRs coming from forks?

**Comment by carkom** [comment]
> Will this work for PRs coming from forks? 

I'm not sure. Can we test it somehow?

**Comment by MatissJanis** [comment]
> I'm not sure. Can we test it somehow?

Create a new repository with the action. Fork it. And then submit a PR to it.

**Comment by carkom** [comment]
> > I'm not sure. Can we test it somehow?
> 
> Create a new repository with the action. Fork it. And then submit a PR to it.

I can't get it working. I'm a bit outside my depth here.

**Comment by twk3** [comment]
@carkom the workflow is missing a run line and getting this error:

```
[Error: .github#L1](https://github.com/actualbudget/actual/commit/a21aee7f069fe6375b413948abf4cfe51664bde7#annotation_19076203337)
every step must define a `uses` or `run` key
```

https://github.com/actualbudget/actual/actions/runs/8217141693

**Comment by carkom** [comment]
> @carkom the workflow is missing a run line and getting this error:
> 
> ```
> [Error: .github#L1](https://github.com/actualbudget/actual/commit/a21aee7f069fe6375b413948abf4cfe51664bde7#annotation_19076203337)
> every step must define a `uses` or `run` key
> ```
> 
> https://github.com/actualbudget/actual/actions/runs/8217141693

Yes, indeed. Do you know how to fix it? I'm a bit out of my depth on this one.

**Comment by twk3** [comment]
Simply passing the new new title to `with` isn't enough to have github change the title. You need to run a command/program to perform the change.

Here is a version that works, but still needs some work (not add WIP if already found in the title for example, and maybe we should have some other escape to prevent it from running when we know we don't want it to WIP)

**Comment by carkom** [comment]
Thanks @twk3. Super helpful.

The idea was to add WIP always. What scenarios do you forsee where we'd want to skip this rule?

**Comment by MatissJanis** [comment]
I don't think this would work in a fork as forks don't have access to github secret tokens AFAIK. 

Would you mind sending the link where you tested it?

**Comment by carkom** [comment]
> I don't think this would work in ~a fork~ PRs coming from forks as forks don't have access to github secret tokens AFAIK.
> 
> Would you mind sending the link where you tested it?

https://github.com/carkom/hello-world/pull/3

**Review by MatissJanis — DISMISSED** [review]
Thanks!

**Comment by carkom** [comment]
> Thanks!

Sorry, slight change to syntax change "WIP: " to "[WIP] "

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**